src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -33,9 +37,11 @@ internal class CurlHandler : HttpMessageHandler
 
         internal CurlHandler()
         {
+            Debug.Assert(s_CurlInitialized);
```

Comment:
Rather than CurlHandler having a static bool s_curlInitialized, seems like InitializeCurl should return void and it should be called from an explicit static cctor on CurlHandler.